### ISSUE TITLE
Page number shows NaN after creating production build with Nx.

### ISSUE THREAD

**Issue by shivamalgoworks** [mui/mui-x#3898]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Instead of showing the total number of entries and the current page we are on the **footer dispalys NaN**. This only happens when I use server side pagination.
![image](https://user-images.githubusercontent.com/93613470/153129434-b39ef96b-b256-449c-a297-ddc452810262.png)

### Expected behavior 🤔

It should be like this.
![image](https://user-images.githubusercontent.com/93613470/153129519-5a9f342d-89a1-4b58-953d-f08844a2b711.png)

### Steps to reproduce 🕹

Steps:

1. Use the dataGrid With server side Pagination.
2. Create a production build using Nx build tool.
3. The footer will show NaN instead of showing numbers.
4.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 6.35 GB / 15.75 GB
  Binaries:
    Node: 14.18.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - C:\Program Files\nodejs\yarn.CMD
  Managers:
    pip3: 21.2.4 - C:\Python310\Scripts\pip3.EXE
  Utilities:
    Git: 2.33.0. - C:\Program Files\Git\mingw64\bin\git.EXE
  IDEs:
    VSCode: 1.64.0 - C:\Users\Shivam Bhatiya\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
  Languages:
    Bash: 4.4.23 - C:\Program Files\Git\usr\bin\bash.EXE
    Perl: 5.34.0 - C:\Program Files\Git\usr\bin\perl.EXE
    Python: 3.10.1 - C:\Python310\python.EXE
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (98.0.1108.43)
    Internet Explorer: 11.0.19041.1202
```
</details>

**Comment by shivamalgoworks** [comment]
No, it does not occur if I run it on local host. It only happens when I create production build using Nx.

**Comment by cherniavskii** [comment]
> I checked row count is of type Number.

Well, `typeof NaN` is `number` in JavaScript 🙃
You'll have to track the exact value.
I've created this example simulating what you showed on the screenshot:
https://codesandbox.io/s/pagesizecustomoptions-material-demo-forked-q4nvu?file=/demo.tsx

> Is there any way I can debug after creating Nx build?

I've never used Nx before, but you can give `Source` panel in Chrome dev tools a try.
I would log `rowCount` value to the console - it's probably the fastest way.
